ckanext/versioning/logic/action.py

```diff
@@ -1,22 +1,68 @@
 # encoding: utf-8
+import datapackage
 import difflib
 import json
 import logging
 import re
 from datetime import datetime
 
+
 from ckan import model as core_model
+from ckan_datapackage_tools import converter
 from ckan.logic.action.get import package_show as core_package_show
 from ckan.logic.action.get import resource_show as core_resource_show
+from ckan.logic.action.create import package_create as core_package_create
+from ckan.logic.action.update import package_update as core_package_update
 from ckan.plugins import toolkit
 from sqlalchemy.exc import IntegrityError
-
 from ckanext.versioning.logic import helpers as h
 from ckanext.versioning.model import DatasetVersion
+from metastore.backend.gh import GitHubStorage
 
 log = logging.getLogger(__name__)
 
 
+def _get_github_backend():
+    token = toolkit.config.get('ckanext.versioning.github_token')
+    #TODO: Define how to handle default_owner
+    default_owner = toolkit.config.get('ckanext.versioning.default_owner')
+    backend = GitHubStorage(
+        github_options={"login_or_token": token},
+        default_owner=default_owner)
+    return backend
+
+def package_create(context, data_dict):
+    """Overrides core package create.
+
+    After creating the package, it calls metastore-lib to create a new GitHub
+    repository a store the package dict in a datapackage.json file.
+    """
+    pkg_dict = core_package_create(context, data_dict)
+
+    if data_dict['type'] == 'dataset':
+        datapackage = converter.dataset_to_datapackage(pkg_dict)
+        backend = _get_github_backend()
+        pkg_info = backend.create(pkg_dict['name'], datapackage)
```

Comment:
You now can (and should) pass an Author object to `create` / `update` so Git commits are attributed to the person who made the change.

Reply:
Also you can pass a commit message, I'm just not sure what that might be.